docs/ref/clvm.md

```diff
@@ -603,53 +603,224 @@ When used in an integer context, nil behaves as zero.
+* Time per aggregate signature validation: 0.972140 ms
+* Total time for 2000 key and signature validations: (0.179370 + 0.972140) * 2000 = 2303.02 ms
+
+Each 1 cost is designed to require 1 nanosecond, so we need to multiply the result by 1 million (ms/ns).
+* Cost for the generator program's BLS operations: `2303.02 * 1,000,000 = 2,303,020,000`.
+
+Using this info, we can also calculate the cost of each `AGG_SIG_UNSAFE` and `AGG_SIG_ME` condition in all CLVM programs:
+* Cost per BLS condition: `(0.179370 + 0.972140) * 1,000,000 = 1,151,510`. We roud this number up to 1,200,000.
+
+### Generator program cost
+
+(This is the total cost of 1, above.)
+
+Taking the previous two calculations into account, the total cost to execute and run the BLS operations of the generator program is: `1,317,054,957 + 2,303,020,000 = 3,620,074,957`.
+
+### Generator program size
+
+(This is the cost of 2, above.)
+
+We know that 1, 2, and 3 all will be assigned equal maximum costs, which we've already established is 3,620,074,957. This is the size-based cost of the generator program.
+
+The generator program itself is 298,249 bytes. Each byte, therefore has a cost of `3,620,074,957 / 298,249 = 12,137.76`. We round this number to 12,000 per byte. This is the cost per bye of all CLVM programs.
+
+### Generator program coins
+
+(This is the cost of 3, above.)
+
+Just like the previous calculation, the total cost of the generator program's coins is 3,620,074,957. The generator program creates 2000 coins, so the cost per `CREATE_COIN` in all CLVM programs is `3,620,074,957 / 2000 = 1,810,037.4785`. We round this number to 1,800,000.
+
+### Maximum cost per block
+
+To calculate the maximum cost per block, we simply add the generator program's execution, size, and coin costs:
+
+Theoretical maximum cost per block: `3,620,074,957 + 3,620,074,957 + 3,620,074,957 = 10,860,224,871` We round this number to 11,000,000,000.
+
+### Maximum block size
+
+The theoretical maximum size of a single block is `maximum cost per block / cost per byte`, or `11,000,000,000 / 12,000 = 916,667 bytes`. However, this number ignores the costs of all operators. If you want a CLVM program to do anything useful, the maximum size would be closer to 400 KB.
+
+Even this number is not realistic because it assumes that a single program will take up an entire block. The maximum number of vanilla transactions (with two outputs) per block is 1000. Therefore, if there is fee pressure on Chia's blockchain, a 400 KB program would need to include a larger fee than the top 1000 vanilla transactions in the mempool -- combined -- in order for a farmer to include it.
+
+### Cost tables
+
+The following table shows the two costs that must be applied to all operators, namely a mandatory cost and a cost per byte of memory.
+
+| type                   | base cost   | cost per byte |
+| ---------------------- | ----------- | ------------- |
+| mandatory cost         | 1           | -             |
```

Comment:
just thinking aloud; we might be able to just add 1 to all operator base costs, rather than listing this separately.
The only reason I can think of to list this separately is that the implementation counts up this cost up-front, before evaluating arguments, whereas the other operator cost counts up when executing the operator.

In practice it won't make a difference, since the cost will be the same, and the program either fails or succeeds. But it may fail in slightly different places. Where a program fails isn't observable by the program or anything else in the consensus rules, so it's really a minor point.